### DISCUSSION TITLE
Surface Laptop Studio Touchpad

### DISCUSSION THREAD

**Discussion by ANTUMN0S** [linux-surface/linux-surface#1079] [Q&A]
I just installed Ubuntu 22.04.2 LTS on my SLS and installed the linux-surface kernel. It all seems to work well, but I can't get my head around the quirk I need to add for the touchpad to work properly (it detects the clicks, but I can't move it).
Here's what I tried:

- Simply creating the directory in /etc and the creating the .quirks file with vim
- Manually adding/building libinput in the /etc directory and creating the .quirks file with vim
- adding the same .quirk file to a different directory that was shown when I followed some steps (also using vim)

None of the methods worked and I don't know what else I could do to get it to work. I haven't found anything similar online and the only thing close to it talking about X11, which to my understanding is an alternative to wayward, but not what's delivered with this kernel.

Has anyone experienced something similar?

Cheers

**Comment by eliezerchavez** [comment]
having the exact same issue: touchscreen works fine, sounds works fine, keyboard works fine and touchpad clicks but doesn't move... (this is in both, Wayland and X)

**Comment by eliezerchavez** [comment]
Hi @ANTUMN0S, this solves the problem:

https://github.com/linux-surface/linux-surface/wiki/Surface-Laptop-Studio#touchpad-quirks

**Reply by ANTUMN0S** [reply]
Hey @eliezerchavez , unfortunately this does not work for me. As I mentioned above, I already tried following that guide and it changes nothing for me. I don't know if I set it up the wrong way. But I do have the directory and the .quirks file, it just doesn't do anything for me.

**Reply by qzed** [reply]
Can you check with `sudo libinput quirks list <device>` if the quirk has been applied correctly? You can use `sudo evtest` to find the `/dev/input/eventX` for the touchpad.

**Reply by ANTUMN0S** [reply]
It doesn't seem like it:
quirks error: Unknown key AttrEventCodeDisable in [StarBook Mk V] (30-vendor-starlabs.quirks)
quirks parser error: /usr/share/libinput/30-vendor-starlabs.quirks:4: failed to parse AttrEventCodeDisable=BTN_RIGHT
Failed to initialize the device quirks. Please see the above errors and/or re-run with --verbose for more details

By the way, for me it's dev/input/event24, if that matters.

**Reply by qzed** [reply]
That looks like some issue with your libinput installation. The file where the error occurs is not modified by us. Since you mentioned that you built libwacom yourself, can you try to "reset" it to the system-provided version?

**Reply by ANTUMN0S** [reply]
Sorry for the question, but how would I go about this? I'm still fairly new in the whole Linux space, so I get lost sometimes.

**Reply by qzed** [reply]
If you remember the steps you took when building libinput (and installing it) you can try to undo/revert those. Otherwise, you should try to reinstall the libinput package.

**Reply by ANTUMN0S** [reply]
Sorry for the delay, I was quite busy.
I decided to remove Ubuntu and do a clean install. Now before I mess it up again, I thought I'd ask here again.
I just installed the Surface kernel and haven't done anything else yet. The input device is still event24, but right now, no instance of libinput is installed (and there is no libinput directory present in /etc).
Should I just install libinput with "sudo apt-get install libinput" or is there another route I should take?

**Reply by qzed** [reply]
Yes, try `sudo apt-get install libinput`. Normally, it should already be installed. The `/etc/libinput` directory does not necessarily have to exist on a fresh install, so you may need to create it.

**Reply by ANTUMN0S** [reply]
Okay, so here's an interesting twist. Running `sudo apt-get install libinput` results in the error "E: Unable to locate package libinput". This might also explain why it wasn't installed in the first place. Looking it up on [https://packages.ubuntu.com/search?suite=jammy&searchon=names&keywords=libinput] reveals that there are multiple packages with _libinput_ in the name, which of these should I install instead? My best guess is _libinput-bin_, as it mentions quirks.

**Reply by qzed** [reply]
Yes, I think `libinput-bin` is the one.
